### ISSUE TITLE
Bit-Genie v9 loses on time in CuteChess

### ISSUE THREAD

**Issue by tissatussa** [Aryan1508/Bit-Genie#212]
i successfully compiled your newest v9 source code into a 64-bit Linux binary .. then, playing in CuteChess, the engine loses on time at the very last move of the timecontrol : i gave 8 minutes per 40 moves plus 3 seconds bonus (and 1000 ms margin) .. this happens with both White and Black .. see screenshots below .. the engine seems to calculate exactly 15 sec each move but fails to mention the remaining time ..

is some counter 1 off ?
then it should be an easy fix !

![bit-genie_v9_loses_on_time1](https://user-images.githubusercontent.com/1109281/156760788-26674e44-7619-4ff2-9165-4d015cbab7b9.png)

![bit-genie_v9_loses_on_time2](https://user-images.githubusercontent.com/1109281/156760803-d9ce9490-0d73-42d4-b5e0-d9390a7577f8.png)

**Comment by Aryan1508** [comment]
Is this the latest commit (95c5cc19f721ba1c6a467b77a79c974b37e5a96d), or the previous release? (https://github.com/Aryan1508/Bit-Genie/releases/tag/v9)

**Comment by tissatussa** [comment]
i just did `git clone https://github.com/Aryan1508/Bit-Genie.git` and compiled ..

**Comment by Aryan1508** [comment]
Alright, that must the latest commit and not v9.
Thanks for the information, I'm looking into it

**Comment by tissatussa** [comment]
after the command `uci` i see :
`id name Bit-Genie 9`

**Comment by Aryan1508** [comment]
Can you send the full uci output?

**Comment by tissatussa** [comment]
```
$ ./Bit-Genie_v9_compiled_HP 
Bit-Genie by Aryan Parekh
uci
id name Bit-Genie 9
id author Aryan Parekh
option name Hash type spin default 8 min 2 max 3000
option name Clear Hash type button
option name OwnBook type check default false
option name BookPath type string
uciok
```

**Comment by Aryan1508** [comment]
Okay, this is v9 and not the latest source.

**Comment by tissatussa** [comment]
OK, point me to the latest / good version .. is the 'normal' git clone version not OK ?

**Comment by Aryan1508** [comment]
When did you clone it?

**Comment by tissatussa** [comment]
about 4 hours ago

**Comment by tissatussa** [comment]
and you could add a 9.x sub-numbering !?

**Comment by Aryan1508** [comment]
```
git clone https://www.github.com/Aryan1508/Bit-Genie
cd Bit-Genie/src
make
./Bit-Genie
```
should give
```
id name Bit-Genie 9.23
id author Aryan Parekh
id network 139cc2fa
option name Hash type spin default 8 min 2 max 262144
option name Threads type spin default 1 min 1 max 1024
option name Clear Hash type button
option name OwnBook type check default false
option name BookPath type string
uciok
```

**Comment by tissatussa** [comment]
exactly !  I just did these commands and the outcome is just like that !  Does `git clone https://www.github.com/Aryan1508/Bit-Genie` differ from `git clone https://www.github.com/Aryan1508/Bit-Genie.git` as i mentioned ? - with(out) the '.git' ? Or did you change the master code in the meantime ?

**Comment by Aryan1508** [comment]
are you sure you're running the correct executable after make? (`Bit-Genie`)

**Comment by tissatussa** [comment]
my first compiled version (4 hours ago) is 522 Kb and the second (v9.23) is 1.6 Mb

**Comment by tissatussa** [comment]
your makefile creates only 1 binary, called 'Bit-Genie' ..

**Comment by Aryan1508** [comment]
you were using probably using an old compile, the last change to master was made over a month ago.
You can continue to use 9.23 as the latest Bit-Genie version. I will commit a fix for the issue in a bit

**Comment by tissatussa** [comment]
i'm on Xubuntu 21.10

**Comment by tissatussa** [comment]
thanks. I tried to describe what i did and what happened, as good as i'm able to .. i'm confused about these different versions .. anyhow, i have a working version 9.23 now ..

**Comment by tissatussa** [comment]
in CuteChess (newest v1.2.0) the compiled Bit-Genie v9.23 also loses on time :

![bit-genie_v9_loses_on_time3](https://user-images.githubusercontent.com/1109281/156805458-248206e9-bf98-4983-b474-69547754c242.png)

**Comment by Aryan1508** [comment]
Fixed, #213
